### PR TITLE
docs: rewrite LightlyTrain tutorial around distillation on CUB-200

### PR DESCRIPTION
**Pull request by gabrielfruet**
## What has changed and why?

The tutorial adapted DINOv2 on Imagenette, an ImageNet subset where DINOv2 already sat at its ceiling, so adaptation moved nothing measurable. The rewrite moves to CUB-200-2011: 11,788 photos, 200 bird species, where fine-grained recognition makes capacity matter. Distilling a `dinov3/vitb16` teacher into an already-distilled `dinov3/vitt16` student (Lightly's, not Meta's) over 100 epochs, about 21 minutes on two RTX 4090s, lifts kNN class purity from 0.081 to 0.606. The teacher scores 0.715. Exported weights inherit the DINOv3 license, which the tutorial now states.

Review found the example scripts unrunnable: the CUB host answers urllib's default user agent with 403. The download and extract guards also tested existence rather than completion, `pretrain` ran without an `if __name__ == "__main__":` guard and crashed on macOS, and the explore script inferred the folder layout and quietly built a different dataset. It now declares one supported layout and raises otherwise.

## How has it been tested?

ruff, mypy and `mkdocs build --clean --strict` pass; `pytest -n auto` gives 2320 passed, 43 skipped. The tutorial's code blocks concatenate and pass `ruff check --select F`. A live probe of the CUB endpoint returns 200. Fixture checks confirmed the extraction sentinel, both explore-script guards and the species labels.

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

Review: @IgorSusmelj, backup @michal-lightly.

## Summary by CodeRabbit

* **Documentation**
  * Updated the embeddings tutorial to demonstrate distilling a large DINOv3 model into a smaller student model using the CUB-200-2011 bird dataset.
  * Added guidance on distillation benefits, model selection, hardware requirements, result exploration, and curation workflows.
  * Refreshed tutorial visuals, descriptions, and performance comparisons.

* **Examples**
  * Updated training and exploration examples for teacher-student distillation and species-based labels.
  * Added configurable species and image limits for exploration.
  * The workflow now exports and explores the distilled student embedding model.

### CODE REVIEW

File: lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md [hunk 1]
```diff
@@ -252,63 +280,70 @@ Add one line to the end of `explore.py` to open the app:
...
+Click the `Embed` button in the top right to open the [embedding plot](../concepts_and_tools/embeddings.md#the-embedding-plot-gui). The plot shows every image as a point in a 2D projection (PaCMAP) of the embedding space. To color the points by species, open **Color by** at the bottom of the plot and pick **annotations**.
 
 Read the map:
 
 - **Hover a point** to preview its image.
-- **Look for clusters** — tight groups are visually similar images; a well-trained model separates your categories into distinct blobs.
-- **Watch the edges** — points far from any cluster are outliers or mislabeled data worth a look.
-- **Spot near-duplicates** — points stacked on top of each other are almost identical images.
+- **Look for clusters.** Tight groups are visually similar images.
+- **Watch the edges.** Points far from any cluster are outliers or wrong labels.
```
[the remark this answers is not in the document]
Comment (gabrielfruet) [reply lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md]: Fixed in cc53f7b: "wrong labels" is now "wrong annotations". The remaining uses of "label" describe training without ground truth ("uses no labels", "unlabeled images"), which is SSL terminology for LightlyTrain rather than an annotation on a sample, so those stay.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md [hunk 2]
```diff
@@ -1,54 +1,74 @@
...
-    <figcaption><strong>A generic model (MobileCLIP).</strong> Classes overlap and bleed together.</figcaption>
+    <img src="https://storage.googleapis.com/lightly-public/studio/tutorials/lightlytrain-embeddings/dinov3-tiny-before.png" alt="Embeddings from an off-the-shelf DINOv3 ViT-T, colored by species" style="width: 100%; border-radius: 6px;">
+    <figcaption><strong>Off the shelf (purity 0.343).</strong> The species overlap.</figcaption>
   </figure>
   <figure style="margin: 0;">
-    <img src="https://storage.googleapis.com/lightly-public/studio/tutorials/lightlytrain-embeddings/dinov2-embeddings.jpg" alt="LightlyTrain embeddings colored by class" style="width: 100%; border-radius: 6px;">
-    <figcaption><strong>Your LightlyTrain model.</strong> The same classes separate into distinct clusters.</figcaption>
+    <img src="https://storage.googleapis.com/lightly-public/studio/tutorials/lightlytrain-embeddings/dinov3-tiny-after-distill.png" alt="Embeddings from a distilled DINOv3 ViT-T, colored by species" style="width: 100%; border-radius: 6px;">
+    <figcaption><strong>After distillation (purity 0.886).</strong> The species separate.</figcaption>
```
[the remark this answers is not in the document]
Comment (gabrielfruet) [reply lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md:49]: Fixed in cc53f7b. Added a paragraph directly above the figures: the caption purity is measured on the 978 images from 20 species, the table measures all 200 species, and fewer species is an easier task, so the caption values are higher. Same metric in both. The 20-species slice is what `explore.py` now loads, so the figures are reproducible.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: lightly_studio/src/lightly_studio/examples/example_lightlytrain_train_and_export.py [hunk 3]
```diff
@@ -20,30 +28,37 @@
...
-archive = Path("imagenette2-320.tgz")
+# 1. Download CUB-200-2011 (11,788 images of 200 bird species), one folder per species.
+archive = Path("CUB_200_2011.tgz")
 if not archive.exists():
     urllib.request.urlretrieve(
-        "https://s3.amazonaws.com/fast-ai-imageclas/imagenette2-320.tgz", archive
+        "https://data.caltech.edu/records/65de6-vp158/files/CUB_200_2011.tgz", archive
     )
     with tarfile.open(archive) as tar:
         tar.extractall(".")
```
[the remark this answers is not in the document]
Comment (gabrielfruet) [reply lightly_studio/src/lightly_studio/examples/example_lightlytrain_train_and_export.py]: Fixed in cc53f7b: `tar.extractall(".", filter="data")` in both the example and the tutorial snippet, so a crafted archive cannot write outside the target directory.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md [hunk 4]
```diff
@@ -60,46 +82,61 @@ pip install lightly-train lightly-studio
...
+archive = Path("CUB_200_2011.tgz")
 if not archive.exists():
     urllib.request.urlretrieve(
-        "https://s3.amazonaws.com/fast-ai-imageclas/imagenette2-320.tgz", archive
+        "https://data.caltech.edu/records/65de6-vp158/files/CUB_200_2011.tgz", archive
     )
     with tarfile.open(archive) as tar:
-        tar.extractall(".")
+        # filter="data" refuses members that would write outside the target directory.
+        tar.extractall(".", filter="data")
```
[the remark this answers is not in the document]
Comment (gabrielfruet) [reply lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md]: Fixed in 8ca9bef: Prerequisites now says 3.10.12+, 3.11.4+, or 3.12+.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md [hunk 5]
```diff
@@ -1,54 +1,77 @@
...
 
-- Train (or adapt) an embedding model on your own images with LightlyTrain.
-- Export the model and run it inside LightlyStudio to embed your data.
+- Distill a large pretrained model into a small one, with LightlyTrain and no labels.
+- Export the small model and run it inside LightlyStudio to embed your data.
 - Explore the 2D embedding plot to find clusters, outliers, and near-duplicates.
 - Select a diverse subset for labeling or training.
 
-![LightlyTrain embeddings in LightlyStudio, colored by class — the clusters separate cleanly](https://storage.googleapis.com/lightly-public/studio/tutorials/lightlytrain-embeddings/dinov2-embeddings.jpg){ width="100%" }
+![Embeddings from a distilled DINOv3 ViT-T, colored by species. The clusters separate cleanly](https://storage.googleapis.com/lightly-public/studio/tutorials/lightlytrain-embeddings/dinov3-tiny-after-distill.png){ width="100%" }
```
Comment (liopeer) [inline lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md]: It's a bit confusing to say `species` here without having yet mentioned the dataset.

File: lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md [hunk 6]
```diff
@@ -1,54 +1,77 @@
...
   <figure style="margin: 0;">
-    <img src="https://storage.googleapis.com/lightly-public/studio/tutorials/lightlytrain-embeddings/dinov2-embeddings.jpg" alt="LightlyTrain embeddings colored by class" style="width: 100%; border-radius: 6px;">
-    <figcaption><strong>Your LightlyTrain model.</strong> The same classes separate into distinct clusters.</figcaption>
+    <img src="https://storage.googleapis.com/lightly-public/studio/tutorials/lightlytrain-embeddings/dinov3-tiny-after-distill.png" alt="Embeddings from a distilled DINOv3 ViT-T, colored by species" style="width: 100%; border-radius: 6px;">
+    <figcaption><strong>After distillation (purity 0.886).</strong> The species separate.</figcaption>
   </figure>
 </div>
 
+!!! note "Pick a dataset where a larger model helps"
+    Distillation moves the knowledge of the teacher into the student. If a large model scores about the same as a small one on your data, there is nothing to move. On CUB-200, ViT-L scores 0.195 above ViT-S, so there is a large margin to recover. On a 10-class satellite dataset, the same margin was only 0.022, and distillation gives almost nothing.
```
Comment (liopeer) [inline lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md]: Would not mention numbers for satellite, since we don't specify the exact dataset etc.

File: lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md [hunk 7]
```diff
@@ -1,54 +1,77 @@
...
 ## Prerequisites
 
-To follow this tutorial, make sure you have:
+To follow this tutorial, make sure that you have:
 
-- Python 3.10 or newer
-- A GPU for the training step. A CPU also works, since `dinov2/vits14` starts from pretrained weights — just slower.
-- The exploration step embeds your images on a CUDA GPU, Apple Silicon (MPS), or a CPU.
-- About 2 GB of free disk space
+- Python 3.10.12+, 3.11.4+, or 3.12+. Step 1 uses the `filter` argument of `tarfile`. Earlier patch releases do not have it.
```
Comment (liopeer) [inline lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md]: That seems oddly fine-grained in terms of Python versions.

File: lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md [hunk 8]
```diff
@@ -60,65 +83,96 @@ pip install lightly-train lightly-studio
...
+        # run there is no checkpoint and this does nothing.
+        resume_interrupted=True,
+    )
 ```
 
-Training writes a checkpoint to `out/pretrain/checkpoints/last.ckpt`. Any name from `lightly_train.list_models()` works for `model` — see LightlyTrain's [supported models](https://docs.lightly.ai/train/stable/pretrain_distill/models/index.html). To train with more data, a different backbone, or another method (DINOv2, DINO, SimCLR), see LightlyTrain's [pretraining guide](https://docs.lightly.ai/train/stable/pretrain_distill/index.html) and [available methods](https://docs.lightly.ai/train/stable/pretrain_distill/methods/index.html).
+Training writes a checkpoint to `out/pretrain/checkpoints/last.ckpt`.
+
+!!! note "Choose the teacher"
+    A larger teacher is not always better. ViT-L has 3.5 times more parameters than ViT-B, so every distillation step costs more, and the teacher is frozen either way. This tutorial did not measure a ViT-T distilled from ViT-L, so the table above cannot tell you what that extra cost buys.
```
Comment (liopeer) [inline lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md]: That's not very helpful imo.

File: lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md [hunk 9]
```diff
@@ -60,65 +83,96 @@ pip install lightly-train lightly-studio
...
 
 ```python title="train_and_export.py"
-lightly_train.export(
-    out="out/embedding_model.pt",
-    checkpoint="out/pretrain/checkpoints/last.ckpt",
-    part="embedding_model",
-    format="torch_model",
-    overwrite=True,
-)
+    lightly_train.export(
```
Comment (liopeer) [inline lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md:156]: Why the indentation?

File: lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md [hunk 10]
```diff
@@ -253,63 +329,72 @@ Add one line to the end of `explore.py` to open the app:
...
 python explore.py            # embed your data and open LightlyStudio
 ```
 
-Click the `Embed` button in the top right to open the [embedding plot](../concepts_and_tools/embeddings.md#the-embedding-plot-gui). It shows every image as a point in a 2D projection (PaCMAP) of the embedding space. To color points by their class, open **Color by** at the bottom of the plot and pick **annotations**.
+Click the `Embed` button in the top right to open the [embedding plot](../concepts_and_tools/embeddings.md#the-embedding-plot-gui). The plot shows every image as a point in a 2D projection (PaCMAP) of the embedding space. To color the points by species, open **Color by** at the bottom of the plot and pick **annotations**.
 
-![The embedding plot after loading LightlyTrain embeddings, colored by class](https://storage.googleapis.com/lightly-public/studio/tutorials/lightlytrain-embeddings/dinov2-embeddings.jpg){ width="100%" }
+![The embedding plot after the distilled model embedded the 20-species slice, colored by species](https://storage.googleapis.com/lightly-public/studio/tutorials/lightlytrain-embeddings/dinov3-tiny-after-distill.png){ width="100%" }
```
Comment (liopeer) [inline lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md]: Do we really want to show it twice?

**Review by liopeer — COMMENTED** [review]
Overall really nice, a few minor comments!

File: lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md (hunk 5 above)
Comment (gabrielfruet) [reply lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md]: Dropped "species" from the caption; it now reads "one color per class". The dataset is not introduced until 10 lines later.
_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md (hunk 6 above)
Comment (gabrielfruet) [reply lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md]: Removed the 0.022 figure. It now says distillation gives almost nothing on coarse-grained data, with no unnamed dataset behind a number.
_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md (hunk 7 above)
Comment (gabrielfruet) [reply lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md]: Agreed. Now "Python 3.10 or newer", with the constraint as a date: `filter` needs a June 2023 or later patch release on 3.10 and 3.11.
_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md (hunk 8 above)
Comment (gabrielfruet) [reply lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md]: Removed the note. Once corrected it only said the ViT-L-teacher measurement is missing, which is not guidance.
_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md (hunk 9 above)
Comment (gabrielfruet) [reply lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md:156]: It was continuing an `if __name__ == "__main__":` opened in Step 1, which read badly out of context. Each step now writes a top-level function, and the guard is its own block calling all three. LightlyTrain spawns dataloader workers that re-import the file, so it exits 1 on macOS and Windows without it.
_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md (hunk 10 above)
Comment (gabrielfruet) [reply lightly_studio/docs/docs/tutorials/lightlytrain-embeddings.md]: Removed the Step 5 copy. Step 5 now points back to the plot at the top of the page.
_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

**Review by liopeer — APPROVED** [review]
LGTM! Thx for addressing the review!
